src/app/Http/Controllers/BusinessTripController.php

```diff
@@ -35,46 +51,236 @@ public function create() {
+            'place_start' => 'nullable|string|max:200',
+            'place_end' => 'nullable|string|max:200',
+            'datetime_border_crossing_start' => 'nullable|date',
+            'datetime_border_crossing_end' => 'nullable|date',
+            'trip_purpose_id' => 'required|integer|min:0', //possibly |exists:trip_purposes,id
+            'purpose_details' => 'nullable|string|max:50',
+            'iban' => 'required|string|max:34',
+            'conference_fee_id' => 'nullable|integer|min:0',
+            'reimbursement_id' => 'nullable|integer|min:0',
+            'spp_symbol_id' => 'nullable|exists:spp_symbols,id',
+            'accommodation_expense_id' => 'nullable|integer|min:0',
+            'travelling_expense_id' => 'nullable|integer|min:0',
+            'other_expense_id' => 'nullable|integer|min:0',
+            'advance_expense_id' => 'nullable|integer|min:0',
+            'not_reimbursed_meals' => 'nullable|string|max:255',
+            'meals_reimbursement' => 'boolean',
+            'expense_estimation' => 'nullable|string|max:20',
+            'cancellation_reason' => 'nullable|string|max:1000',
+            'note' => 'nullable|string|max:5000',
+            'conclusion' => 'nullable|string|max:5000',
+        ]);
 
+        // Set the type of trip based on the selected country
+        $selectedCountry = $request->input('country');
+        $validatedData['type'] = $selectedCountry === 'Slovensko' ? TripType::DOMESTIC : TripType::FOREIGN;
+
+        //Logic to store the trip based on the validated data
+        $trip = new BusinessTrip($validatedData);
+
+        //Handle file uploads
+        if ($request->hasFile('file-upload-id')) {
+            $file = $request->file('file-upload-id');
+
+            //Store the file in the storage/app/trips directory
+            $upload_name = $file->storeAs('trips', 'trip_' . $trip->id . '.' . $file->extension());
+
+            //Save the file path in the model
+            $trip->upload_name = $upload_name;
+
+            //Save the model to the DB
+            $trip->save();
+        }
+
+        //Sending mails
+        $message = '';
+        $recipient = 'admin@example.com';
+        $viewTemplate = 'emails.new_trip_admin';
+
+        // Create an instance of the SimpleMail class
+        $email = new SimpleMail($message, $recipient, $viewTemplate);
+
+        // Send the email
+        Mail::to($recipient)->send($email);
+
+        //Redirecting to the homepage
+        return redirect()->route('components.homepage');
+
+    }
+
+    /**
+     * Get the attachment from a business trip.
+     */
+    public function getAttachment(BusinessTrip $trip) {
+        // Check if the trip has an attachment
+        if (!$trip->upload_name) {
+            abort(404, 'File not found'); // Or other error handling
+        }
+
+        // Build the file path
+        $filePath = storage_path('app/' . $trip->upload_name);
```

Comment:
Toto chceme robit cez ten `Storage` facade, ako sme sa dohodli, cize 

```suggestion
        $filePath = Storage::disk('uploads')->path($trip->upload_name);
```